tests/api/v1/common/testAssociations.js

```diff
@@ -112,7 +112,38 @@ function testAssociations(path, associations, joiSchema, conf) {
     });
   });
 
-  describe('Checking multiple associations', () => {
+
+  /*
+    These tests fail because of a bug in sequelize.
+    It doesn't properly handle multiple associations when a limit is applied:
+    It selects the attributes before doing the join for the association, which
+    causes an error when the foreign key is not included in the fields.
+
+    One thing we tried:
+      According to Sequelize team when limits are set, they enforce the
+    usage of sub-queries, however, when Generator has multiple
+    associations (ie.: Collectors and User) the sub-query does not
+    expose foreign keys (Generator.createdBy).
+      Sequelize by default will try to create subQuery even when there is
+    no subQuery configured because the duplication flag is true by
+    default.
+      So, flagging duplicating=false makes Sequelize avoid cartesian
+    product not generating sub-queries (further check in:
+    /sequelize/model.js, line 441).
+
+    However, setting "duplicating: false" has side effects that break other
+    functionality; it causes associations to not return all elements.
+    This causes the following tests to fail because only one possibleCollector
+    is returned:
+      "get by key includes associations"
+      "get by key: an association can be specified as a field param"
+      "multiple associations can be specified with key and field params"
+
+    These cases are much more common than the problem we were trying to solve,
+    so it's better to go back to how it was before and re-skip these tests
+    for now.
```

Comment:
I discovered this when updating the tests for the association changes. Previously we weren't testing the case where a generator has multiple possibleCollectors, so we didn't catch this before.